### PR TITLE
feat: [UIE-8194] - DBaaS Upgrades and Maintenance 1

### PR DESCRIPTION
**Pull request by corya-akamai**
## Description 📝
DBaaS Upgrades and Maintenance 1 of 4

## Changes  🔄
List any change relevant to the reviewer.
- Add API to apply security patches (outside maintenance window)
- Refactor database types (alphabetize properties, annotate with JS docs deprecatations and new properties
- Add query for new API
- Add mock data for new properties to factory

## Target release date 🗓️
11/12/24

## As an Author I have considered 🤔

*Check all that apply*

- [x] 👀 Doing a self review
- [x] ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- [x] 🤏 Splitting feature into small PRs
- [x] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [ ] 🧪 Providing/Improving test coverage
- [x] 🔐 Removing all sensitive information from the code and PR description
- [ ] 🚩 Using a feature flag to protect the release
- [ ] 👣 Providing comprehensive reproduction steps
- [ ] 📑 Providing or updating our documentation
- [ ] 🕛 Scheduling a pair reviewing session
- [ ] 📱 Providing mobile support
- [ ] ♿  Providing accessibility support

### CODE REVIEW

File: packages/manager/src/queries/databases/databases.ts [hunk 1]
```diff
@@ -135,6 +136,18 @@ export const useDatabaseMutation = (engine: Engine, id: number) => {
...
 };
 
+export const usePatchDatabaseMutation = (engine: Engine, id: number) => {
+  const queryClient = useQueryClient();
+  return useMutation<void, APIError[], void>({
+    mutationFn: () => patchDatabase(engine, id),
+    onSuccess() {
+      queryClient.invalidateQueries({
+        queryKey: databaseQueries.databases.queryKey,
+      });
```
Comment (bnussman-akamai) [inline packages/manager/src/queries/databases/databases.ts:146]: We should invalidate both the Databases list and the specific database to ensure the cache is up to date.

```suggestion
      queryClient.invalidateQueries({
        queryKey: databaseQueries.databases.queryKey,
      });
      queryClient.invalidateQueries({
        exact: true,
        queryKey: databaseQueries.database(engine, id).queryKey,
      });
```

**Review by bnussman-akamai — COMMENTED** [review]
There is some test flake on `DatabaseResize.test.tsx` that would be nice to fix

**Review by cpathipa — APPROVED** [review]
LGTM!  Approved pending on addressing feedback of @bnussman-akamai

File: packages/manager/src/queries/databases/databases.ts (hunk 1 above)
Comment (corya-akamai) [reply packages/manager/src/queries/databases/databases.ts:146]: Done

**Comment by corya-akamai** [comment]
> DatabaseResize.test.tsx

@mpolotsk-akamai is this test fix in your PR https://github.com/linode/manager/pull/11180?

 FAIL  src/features/Databases/DatabaseDetail/DatabaseResize/DatabaseResize.test.tsx > database resize > on rendering of page and isDatabasesV2GA is true and the Dedicated CPU tab is preselected > should render node selection for dedicated tab with default summary
TestingLibraryElementError: Unable to find an element by: [data-testid="database-node-2"]

**Review by bnussman-akamai — APPROVED** [review]

**Pull request merged**